tests/integration/targets/nd_site/tasks/main.yml

```diff
@@ -0,0 +1,246 @@
+    url: '{{ apic_ip_address }}'
+    site_username: '{{ site_username }}'
+    site_password: '{{ site_password }}'
+    site_name: aci_site
+    site_type: aci
+    latitude: 50.887318
+    longitude: 4.447084
+    login_domain: "DefaultAuth"
+    inband_epg: In-Band-EPG
+    state: present
+  check_mode: True
+  register: cm_add_site
+
+- name: Verify cm_add_site
+  assert:
+    that:
+    - cm_add_site is changed
+    - cm_add_site.current.spec.siteType == "ACI"
+    - cm_add_site.current.spec.name == "aci_site"
+    - cm_add_site.current.spec.latitude == "50.887318"
+    - cm_add_site.current.spec.longitude == "4.447084"
+
+# ADD SITE (Type - ACI)
+- name: Create site (normal mode)
+  cisco.nd.nd_site:
+    <<: *site_present
+    state: present
+  register: nm_add_site
+
+- name: Verify create site
+  assert:
+    that:
+    - nm_add_site is changed
+    - cm_add_site is changed
+    - nm_add_site.previous == cm_add_site.previous == {}
+    - nm_add_site.current.spec.siteType == cm_add_site.current.spec.siteType == "ACI"
+    - nm_add_site.current.spec.name == cm_add_site.current.spec.name == "aci_site"
+    - nm_add_site.current.spec.latitude == cm_add_site.current.spec.latitude == "50.887318"
+    - nm_add_site.current.spec.longitude == cm_add_site.current.spec.longitude == "4.447084"
+
+# ADD SITE again
+- name: ADD site again
+  cisco.nd.nd_site:
+    <<: *site_present
+  register: add_site_again
+
+- name: Verify add_site_again
+  assert:
+    that:
+    - add_site_again is not changed
+    - add_site_again.current.spec.siteType == "ACI"
+    - add_site_again.current.spec.name == "aci_site"
+    - add_site_again.current.spec.latitude == "50.887318"
+    - add_site_again.current.spec.longitude == "4.447084"
+
+# MODIFY SITE
+- name: Modify site
+  cisco.nd.nd_site:
+    <<: *site_present
+    latitude: 51.887318
+    longitude: 5.447084
+    state: present
+  register: modify_site
+
+- name: Verify modify_site
+  assert:
+    that:
+    - modify_site is changed
+    - modify_site.current.spec.siteType == "ACI"
+    - modify_site.current.spec.name == "aci_site"
+    - modify_site.current.spec.latitude == "51.887318"
+    - modify_site.current.spec.longitude == "5.447084"
+
+# ADD SITE (Type - Cloud)
+- name: Create Cloud site
+  cisco.nd.nd_site:
+    <<: *nd_info
+    url: '{{ apic_ip_address }}'
+    site_username: '{{ site_username }}'
+    site_password: '{{ site_password }}'
+    site_name: cloud_site
+    site_type: cloud_aci
+    latitude: 50.887318
+    longitude: 4.447084
+    login_domain: "DefaultAuth"
+    inband_epg: In-Band-EPG
+    state: present
+  register: nm_add_cloud_site
+
+- name: Verify nm_add_cloud_site
+  assert:
+    that:
+    - nm_add_cloud_site is changed
+    - nm_add_cloud_site.current.spec.siteType == "CloudACI"
+    - nm_add_cloud_site.current.spec.name == "cloud_site"
+
+# ADD SITE (Type - NDFC)
+- name: Create NDFC site
+  cisco.nd.nd_site:
+    <<: *nd_info
+    url: '{{ apic_ip_address }}'
```

Comment:
Shouldn't apic_ip_address in url: '{{ apic_ip_address }}' be a variable corresponding to the NDFC IP address in the inventory?